mp/src/game/server/momentum/mom_system_saveloc.cpp

```diff
@@ -436,13 +436,24 @@ SavedLocation_t* CSaveLocSystem::CreateSaveloc(int components /*= SAVELOC_ALL*/)
 
 void CSaveLocSystem::CreateAndSaveLocation()
 {
-    SavedLocation_t *created = CreateSaveloc();
-    if (created)
+    const auto pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (!pPlayer)
+        return;
+
+    const auto pEntData = pPlayer->GetRunEntData();
+    if (!pEntData)
+        return;
+
+    if (pEntData->m_bIsInZone && pEntData->m_iCurrentZone == 1)
     {
-        AddSaveloc(created);
-        ClientPrint(CMomentumPlayer::GetLocalPlayer(), HUD_PRINTTALK, CFmtStr("Saveloc #%i created!", m_rcSavelocs.Count()));
-        UpdateRequesters();
+        engine->ServerCommand("mom_start_mark_create\n");
+        UTIL_HudHintText(pPlayer, "#MOM_Hint_CreateStartMark");
+        return;
```

Comment:
I think you can use the direct call to `CreateStartMark` from the `pPlayer` you got above. If it returns true, do the hud hint and return. Otherwise if false, it can fall through and create a normal saveloc as a backup.

Reply:
As above, was just a workaround to get the clientprint without adding a suppressing parameter to the clear function.

Reply:
This is a separate thing, though, with the hud hint. It's fine being here, but creating the start mark could fail and it'll do the hint and return anyways. I was suggesting manually calling CreateStartMark and if true, show the hint and return

Reply:
Oh I gotcha. Yeah the parameter should fix that then